### PR TITLE
WT-10619 Update Clang-Format to 12.0.1

### PR DESCRIPTION
**Pull request by etienneptl**
- We do not download clang format as part of s_clang_format.sh anymore
- We rely on what's installed on the system and expect a specific version
- The current expected version is 12.0.1 and after running it, a few files are impacted by some format changes.

### CODE REVIEW

File: dist/s_clang_format [hunk 1]
```diff
@@ -43,17 +20,12 @@ cd "$dest_dir" || exit 1
 export PATH="${PWD}/dist":$PATH
 
 # Check if Clang-Format is already available with the desired version.
-desired_version="10.0.0"
-if ! command -v clang-format &> /dev/null; then
-    download_clang_format $desired_version || exit 1
-elif ! clang-format --version | grep -q $desired_version; then
-    download_clang_format $desired_version || exit 1
+desired_version="12.0.1"
+if ! command -v clang-format &> /dev/null || ! clang-format --version | grep -q $desired_version; then
```
Comment (lukech) [inline dist/s_clang_format:24]: For better readability, can we use square brackets to group the 2 x conditional checkings (delimited by `||`)?

File: dist/s_clang_format [hunk 2]
```diff
@@ -43,17 +20,12 @@ cd "$dest_dir" || exit 1
...
+    echo "Failed to find Clang-Format, make sure the binary (version: $desired_version) is in your env PATH."
```
Comment (lukech) [inline dist/s_clang_format:25]: Clang-Format --> clang-format (to better match the binary name)? 
As there are 2 conditions covered by the same error message, is it worth mentioning install the package (besides putting the binary under env PATH)?

File: dist/s_clang_format (hunk 1 above)
Comment (etienneptl) [reply dist/s_clang_format]: I divided the checks in 1037fe0fabd228c0acf913d81c977eba7bde8052

File: dist/s_clang_format (hunk 2 above)
Comment (etienneptl) [reply dist/s_clang_format]: Let me know if 1037fe0fabd228c0acf913d81c977eba7bde8052 works for you

File: test/evergreen.yml [hunk 3]
```diff
@@ -4846,6 +4846,7 @@ buildvariants:
     - name: ".pull_request !.pull_request_compilers"
     - name: ".unit_test_long"
     - name: compile
+    - name: s-all
```
Comment (lukech) [inline test/evergreen.yml:4849]: The task has been tagged as "pull_request", and should have been covered by the above line 4846.

Comment (lukech) [reply test/evergreen.yml:4849]: Never mind. Just realized you removed the "pull_request" tag on the task. I guess you want the task only runs on the `ubuntu2004` build, which feels reasonable.

**Review by lukech — APPROVED** [review]
LGTM, thanks.

File: test/evergreen.yml (hunk 3 above)
Comment (etienneptl) [reply test/evergreen.yml:4849]: Yes, that's correct!

**Pull request closed without merging**